public/pages/HistoricalDetectorDetail/containers/HistoricalDetectorDetail/HistoricalDetectorDetail.tsx

```diff
@@ -0,0 +1,410 @@
+  const [
+    historicalDetectorDetailModalState,
+    setHistoricalDetectorDetailModalState,
+  ] = useState<HistoricalDetectorDetailModalState>({
+    isOpen: false,
+    action: undefined,
+  });
+  const callout = getCallout(detector, isStoppingDetector);
+
+  useEffect(() => {
+    if (errorGettingDetectors) {
+      core.notifications.toasts.addDanger(
+        'Unable to get the historical detector'
+      );
+      props.history.push('/historical-detectors');
+    }
+  }, [errorGettingDetectors]);
+
+  useEffect(() => {
+    if (detector) {
+      core.chrome.setBreadcrumbs([
+        BREADCRUMBS.ANOMALY_DETECTOR,
+        BREADCRUMBS.HISTORICAL_DETECTORS,
+        { text: detector ? detector.name : '' },
+      ]);
+    }
+  }, [detector]);
+
+  const fetchDetector = async () => {
+    try {
+      await dispatch(getDetector(detectorId));
+      setIsLoadingDetector(false);
+    } catch {
+      core.notifications.toasts.addDanger(
+        'Error retrieving the historical detector'
+      );
+    }
+  };
+
+  // Try to get the detector initially
+  useEffect(() => {
+    if (detectorId) {
+      setIsLoadingDetector(true);
+      fetchDetector();
+    }
+  }, []);
+
+  // If detector is initialiazing or running: keep fetching every second to quickly update state/results/percentage bar, etc.
+  useEffect(() => {
+    if (
+      detector?.curState === DETECTOR_STATE.RUNNING ||
+      detector?.curState === DETECTOR_STATE.INIT
+    ) {
+      const intervalId = setInterval(
+        fetchDetector,
+        HISTORICAL_DETECTOR_RESULT_REFRESH_RATE
+      );
+      return () => {
+        clearInterval(intervalId);
+      };
+    }
+  }, [detector]);
+
+  const handleDeleteAction = () => {
+    setHistoricalDetectorDetailModalState({
+      ...historicalDetectorDetailModalState,
+      isOpen: true,
+      action: HISTORICAL_DETECTOR_ACTION.DELETE,
+    });
+  };
+
+  const handleEditAction = () => {
+    detector?.curState === DETECTOR_STATE.RUNNING ||
+    detector?.curState === DETECTOR_STATE.INIT
+      ? setHistoricalDetectorDetailModalState({
+          ...historicalDetectorDetailModalState,
+          isOpen: true,
+          action: HISTORICAL_DETECTOR_ACTION.EDIT,
+        })
+      : props.history.push(`/historical-detectors/${detectorId}/edit`);
+  };
+
+  const onStartDetector = async () => {
+    try {
+      // Check to make sure we can start. Current historical detector limit: 2 running per node
+      const clusterStatsResponse = await dispatch(getClusterStats());
+      const nodeCount = get(
+        clusterStatsResponse,
+        'response.nodes.count.data',
+        1
+      );
+      const historicalDetectorsResponse = await dispatch(
+        getHistoricalDetectorList(GET_ALL_DETECTORS_QUERY_PARAMS)
+      );
+      const runningCount = getRunningHistoricalDetectorCount(
+        get(historicalDetectorsResponse, 'response.detectorList')
+      );
+
+      if (nodeCount * 2 <= runningCount) {
```

Comment:
Max running historical detector per node is a dynamic setting. Default value is 2, but user can change it to other values like 10. The start detector API will check if task executing exceeds limit or not.

Reply:
I see. So to clarify, I can remove all logic related to error handling on the frontend for this, and instead just propagate the backend error message?